opslevel/resource_opslevel_check_tool_usage.go

```diff
@@ -229,11 +229,11 @@ func (r *CheckToolUsageResource) ValidateConfig(ctx context.Context, req resourc
 		"tool_url_predicate":    configModel.ToolUrlPredicate,
 	}
 	for predicateSchemaName, predicate := range checkToolUsagePredicates {
-		predicateModel, diags := PredicateObjectToModel(ctx, predicate)
-		resp.Diagnostics.Append(diags...)
-		if predicateModel.Type.IsUnknown() || predicateModel.Type.IsNull() {
+		if predicate.IsNull() || predicate.IsUnknown() {
```

Comment:
Is this check necessary given that `Validate()` will look for this? https://github.com/OpsLevel/terraform-provider-opslevel/pull/409/files#diff-96ab223662412d6624c6efb456fc4cfdd713cdd13ebd3c53140bea84517a4a48R96

Reply:
It is necessary. Here the `predicate` (object) itself is being checked whereas the `Validate()` function is checking if the predicate type field is null or unknown